include/picongpu/particles/filter/RelativeGlobalDomainPosition.def

```diff
@@ -19,6 +19,9 @@
 
 
 #pragma once
+#include "picongpu/particles/traits/SpeciesEligibleForSolver.hpp"
+
+#include <pmacc/traits/HasIdentifiers.hpp>
```

Comment:
What is the reason for moving the trait imlementation to `.def` file?

Including `.hpp` files into `.def` files is potentially dangerous as some `.hpp` files include whole simulation and that may cause wrong inclusion order. So without a good reason we try to avoid doing it.

Reply:
We already do this in files like `Density.def` in `derivedAttributes`. I  got some compile errors when I tried to include the `filter.hpp`in `FilteredDerivedAttribute.hpp`. Some things where not defined yet.   Btw. no when I look at this  `FilteredDerivedAttribute.hpp` could probably also be `.def`.

Reply:
@psychocoderHPC could you comment on that? I personally try to keep `.def` contents (including what they include) at bare minimum, but in this case I don't think it's wrong to do.

Reply:
Moving the traits into the `def` file should be OK. As @pordyna wrote it could be a requirement to have it in the def file because this file is using the traits. 
Traits should not pull critical code e.g. solver implementations into the `def` file.